python/ray/data/datasource/tfrecords_datasource.py

```
@@ -92,25 +122,30 @@ def _convert_arrow_table_to_examples(
 
 def _get_feature_value(
     feature: "tf.train.Feature",
+    schema_feature_type: Optional["schema_pb2.FeatureType"] = None,
 ) -> "pyarrow.Array":
     import pyarrow as pa
-
-    values = (
-        feature.HasField("int64_list"),
-        feature.HasField("float_list"),
-        feature.HasField("bytes_list"),
-    )
+    from tensorflow_metadata.proto.v0 import schema_pb2
+
+    detected_feature_type = {
+        "bytes": feature.HasField("bytes_list")
```

IMO, if a schema is provided, it should be the source of truth. We should ignore the type inferred from the data or even throw an error if the types don't match. WDYT?

---

Both options make sense to me. Should we emit a warning, but not throw an error if the types mismatch (and use the schema-specified type over the inferred type)? @clarkzinzow @c21 thoughts here?

---

IMO we should throw an error as this means the provided schema is inconsistent with actual schema in file.